cbqueryx/queryrespreader.go

```diff
@@ -204,6 +204,42 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 }
 
 func createResourceError(msg string, cause error) ResourceError {
+	err := ResourceError{
+		Cause: cause,
+	}
+
+	if errors.Is(err, ErrScopeNotFound) || errors.Is(err, ErrCollectionNotFound) {
+		parseResourceNotFoundMsg(&err, msg)
```

Comment:
I think it's negligible, but I did write it so I'm a bit biased. The main reason I did it like this was so that we could have
```
err := ResourceError{
		Cause: cause,
	}
``` 
in just one place, if they return the error, I think they'd need to take the cause and construct it themselves. But this isn't a crucial reason so if it read awkwardly I can change it.